### PR TITLE
Added and re-organized JSON items

### PR DESCRIPTION
**Pull request by E3V3A**
- added ESTIMATED radar type (disabled)
- added missing JSON items: squawk,climb,timestamp,glider,ground
- renamed `modeSCode` to `modes` to match lower-case use of other items
- changed order of items to match JSON
- added placeholders for: maxage, stats, ems

### CODE REVIEW

**Comment by E3V3A** [comment]
This would fix #14.

**Comment by derhuerst** [comment]
Note: This PR contains many changes. In the future, please take the time to split your proposals into individual PRs, so people can discuss them separately.

----

> - renamed `modeSCode` to `modes` to match lower-case use of other items

Please keep it as `modeSCode`, for two reasons:

- the other items use [camelCase](https://en.wikipedia.org/wiki/Camel_case)
- don't mix breaking changes with other proposals in one PR

> - changed order of items to match JSON

What does this mean?

> - added placeholders for: maxage, stats, ems

No placeholders please. Only add code that actually does things.

File: lib/radar.js [hunk 1]
```diff
@@ -12,14 +12,18 @@ const headers = {
...
-	FAA: true, // use US/Canada radar data source
-	FLARM: true, // use FLARM data source
-	MLAT: true, // use MLAT data source
-	ADSB: true, // use ADS-B data source
-	inAir: true, // fetch airborne aircraft
-	onGround: false, // fetch (active) aircraft on ground
-	inactive: false, // fetch inactive aircraft (on ground)
-	gliders: false // fetch gliders
+	// see: https://www.flightradar24.com/how-it-works
+	FAA:   		true, 	// use US/Canada radar data source
```
Comment (derhuerst) [inline lib/radar.js:16]: Make the key [camel-cased](https://en.wikipedia.org/wiki/Camel_case). Same for `FLARM`, `MLAT`, `ADSB`, `ESTIMATED`.

File: lib/radar.js [hunk 2]
```diff
@@ -12,14 +12,18 @@ const headers = {
...
+	FLARM: 		true, 	// use FLARM data source
+	MLAT: 		true, 	// use MLAT data source
+	ADSB: 		true, 	// use ADS-B data source
+	ESTIMATED: 	false,	// use out of coverage position estimate if [known|unknown] destination: [2 hr | 10 min] 
+	inAir: 		true, 	// fetch airborne aircraft
+	onGround: 	false, 	// fetch (active) aircraft on ground
+	inactive: 	false, 	// fetch (active) vehicles on ground
+	gliders: 	true 	// fetch gliders
+	// stats: 	true, 	//
```
Comment (derhuerst) [inline lib/radar.js:25]: Remove unused code.

File: lib/radar.js [hunk 3]
```diff
@@ -31,29 +35,27 @@ const request = (north, west, south, east, when, opt = {}) => {
...
-		air: opt.inAir ? '1' : '0',
-		gnd: opt.onGround ? '1' : '0',
-		vehicles: opt.inactive ? '1' : '0',
-		gliders: opt.gliders ? '1' : '0'
-		// todo: bounds, estimated, maxage, stats, history, prefetch
+		bounds: [north, south, west, east].join(','),	// 
+		callback: 'jsonp',						// 
+		// options: altitude, estimated, maxage, speed, stats, ems, (history, prefetch)?
+		// ToDo: 	maxage, stats, (history, prefetch)?
+		faa: opt.FAA ? '1' : '0',				//
```
Comment (derhuerst) [inline lib/radar.js:42]: Remove unnecessary slashes. Keep the diff as small as possible.

File: lib/radar.js [hunk 4]
```diff
@@ -31,29 +35,27 @@ const request = (north, west, south, east, when, opt = {}) => {
...
 
 	const url = endpoint + '?' + qs.stringify(query)
-	return fetch(url, {
-		mode: 'cors',
-		redirect: 'follow',
-		headers,
-		referrer: 'no-referrer',
-		referrerPolicy: 'no-referrer'
-	})
+	return fetch(url, {mode: 'cors', redirect: 'follow', headers})
```
Comment (derhuerst) [inline lib/radar.js:58]: There is a reason why this is in place. Keep it as before and propose changes in another PR.

File: lib/radar.js [hunk 5]
```diff
@@ -71,25 +73,27 @@ const request = (north, west, south, east, when, opt = {}) => {
...
-				modeSCode: d[0] || null,
-				radar: d[7]
+				id,							// Unique F24 id ?
+				modes: d[0] || null,		// Mode-S Transponder code: 6-digit [hex] is a 24-bit ICAO issued code
+				latitude: d[1],				// in decimal degrees
+				longitude: d[2],			// in decimal degrees
+				bearing: d[3], 				// in [degrees]
+				altitude: d[4], 			// in [feet] 
+				speed: d[5], 				// in [knots]
+				squawk: d[6],				// Mode-3/A Transponder code: 4-digit [octal] is the "Squawk" ATC assigned code
```
Comment (derhuerst) [inline lib/radar.js:83]: Use `squawkCode` [as discussed](https://github.com/derhuerst/flightradar24-client/issues/14#issuecomment-363597195).

File: lib/radar.js [hunk 6]
```diff
@@ -71,25 +73,27 @@ const request = (north, west, south, east, when, opt = {}) => {
...
+				radar: d[7],				// F24 "radar" data source (ADS-B, MLAT, RADAR (ATC feed), FLARM, "ESTIMATED") designator
+				model: d[8] || null,		// ICAO Aircraft Type Designator
+				registration: d[9] || null,	// ICAO Aircraft (tail) registration number
+				timestamp: d[10] || null, 	// Timestamp [Unix/POSIX/epoch]
+				origin: d[11] || null,		// Departure Airport IATA
+				destination: d[12] || null,	// Destination Airport IATA
+				flight: d[13] || null,		// IATA Flight Id
+				ground: d[14],				// Airplane is "onGround": [0|1]
```
Comment (derhuerst) [inline lib/radar.js:91]: Use `isOnGround`.

File: lib/radar.js [hunk 7]
```diff
@@ -71,25 +73,27 @@ const request = (north, west, south, east, when, opt = {}) => {
...
+				squawk: d[6],				// Mode-3/A Transponder code: 4-digit [octal] is the "Squawk" ATC assigned code
+				radar: d[7],				// F24 "radar" data source (ADS-B, MLAT, RADAR (ATC feed), FLARM, "ESTIMATED") designator
+				model: d[8] || null,		// ICAO Aircraft Type Designator
+				registration: d[9] || null,	// ICAO Aircraft (tail) registration number
+				timestamp: d[10] || null, 	// Timestamp [Unix/POSIX/epoch]
+				origin: d[11] || null,		// Departure Airport IATA
+				destination: d[12] || null,	// Destination Airport IATA
+				flight: d[13] || null,		// IATA Flight Id
+				ground: d[14],				// Airplane is "onGround": [0|1] 
+				climb: d[15],				// "Rate of Climb" (RoC) is a vertical speed [ft/m]
```
Comment (derhuerst) [inline lib/radar.js:92]: Use `rateOfClimb`.

File: lib/radar.js [hunk 8]
```diff
@@ -71,25 +73,27 @@ const request = (north, west, south, east, when, opt = {}) => {
...
+				callsign: d[16] || null,	// ICAO ATC call signature
+				glider: d[17],				// Aircraft is a glider:   [0|1]  (There are usually no or very few gliders registered on earth!)
```
Comment (derhuerst) [inline lib/radar.js:94]: Use `isGlider`.

**Review by derhuerst — CHANGES_REQUESTED** [review]

**Comment by derhuerst** [comment]
Also, please update the tests and the readme.

**Comment by E3V3A** [comment]
Hi Jannis!

Your proposed changes, essentially reverts my entire PR. In addition, with my very limited (and recently acquired) nodejs skills, I don't think I qualify to update the tests. So since my PR was primarily intended to provide you with the missing details, feel free to implement it at your own convenience. I will not be able to make any further changes.

Best Regards,
E:V:A

**Comment by derhuerst** [comment]
Hey,

note that my proposed changes only revert the part that is out of scope of this PR. I'm willing to accept it as long as it doesn't change this that we discussed before *not to be changed*.

> So since my PR was primarily intended to provide you with the missing details [...]

Thanks again! Thanks to your input in #14, people can now make `flightradar24-client` a more comprehensive tool.

> I will not be able to make any further changes.

Hm, sad to hear that. Would you continue doing this if I updated the tests? (You would still have to implement my proposed changes though.)

**Pull request closed without merging**